frame/contracts/src/wasm/runtime.rs

```diff
@@ -1189,21 +1157,14 @@ define_env!(Env, <E: Ext>,
 				delta.push(delta_key);
 
 				// Offset key_ptr to the next element.
-				key_ptr = key_ptr.checked_add(KEY_SIZE as u32).ok_or_else(|| sp_sandbox::HostError)?;
+				key_ptr = key_ptr.checked_add(KEY_SIZE as u32).ok_or(Error::<E::T>::OutOfBounds)?;
 			}
 
 			delta
 		};
 
-		if let Ok(()) = ctx.ext.restore_to(
-			dest,
-			code_hash,
-			rent_allowance,
-			delta,
-		).map_err(|e| ctx.store_err(e)) {
-			ctx.trap_reason = Some(TrapReason::Restoration);
-		}
-		Err(sp_sandbox::HostError)
+		ctx.ext.restore_to(dest, code_hash, rent_allowance, delta)?;
+		Err(TrapReason::Restoration.into())
```

Comment:
isn't `into` useless ?
```suggestion
		Err(TrapReason::Restoration)
```